<launch>
  <arg name="cam_name" />
  <arg name="calib_file" default="$(find asr_sdm_ros2)/param/calib/$(arg cam_name).yaml"/>

  <!-- SVO node -->
  <node pkg="asr_sdm_ros2" type="svo_node" name="svo" clear_params="true" output="screen" >

    <!-- Camera topic to subscribe to -->
    <param name="cam0_topic" value="camera/image_raw" type="str" />

    <!-- Camera calibration file -->
    <param name="calib_file" value="$(arg calib_file)" />

    <!--Parameters-->
    <rosparam file="$(find asr_sdm_ros2)/param/pinhole.yaml" />
    <param name="map_scale" value="1" />
    
    <!-- Loop closure detection parameters -->
    <param name="runlc" value="false" />
    <!--<rosparam file="$(find svo_ros)/param/loop_closing.yaml" />-->
    <!--<param name="voc_path" value="$(find svo_online_loopclosing)/vocabularies/" type="str" />-->

  </node>

  <!-- RVIZ + SVO GUI -->
  <node name="vis" pkg="rviz" type="rviz" args=" -d $(find asr_sdm_ros2)/rviz_config.rviz" />
  <node name="svo_gui" pkg="rqt_gui" type="rqt_gui" args="-s rqt_svo.svo.Svo --args --topic svo" />

</launch>
